### PR TITLE
feat(dp/pt): refactor `se_e3` descriptor

### PR DESCRIPTION
**Pull request by iProzd**
Note:
1. `exclude_types` is supported only for pt/dp.
2. Note that an exsiting bug in TF is fixed in deepmd/tf/env.py, when `resnet_dt` is `True` for `se_e3`.

## Summary by CodeRabbit

- **New Features**
  - Introduced `DescrptSeT` class for DeepPot-SE descriptor with enhanced atomic configuration handling.
  - Added `DescrptBlockSeT` class for descriptor block functionality.

- **Improvements**
  - Enhanced parameter management and serialization methods in descriptor classes.
  - Added `env_protection` parameter for better environmental control.

- **Bug Fixes**
  - Improved method signatures for better consistency and error handling.

- **Tests**
  - Added comprehensive test cases for `DescrptSeT` class across different deep learning frameworks.
  - Included consistency checks and JIT compilation tests.

### CODE REVIEW

File: examples/water/se_e3/input_torch.json
Comment (njzjz) [inline examples/water/se_e3/input_torch.json:1]: Is there any difference between this and the TF file?

File: source/tests/pt/model/test_se_e2_a.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def test_consistency(
                 precision=prec,
                 resnet_dt=idt,
                 old_impl=False,
-                exclude_mask=em,
+                exclude_types=em,
```
Comment (wanghan-iapcm) [inline source/tests/pt/model/test_se_e2_a.py:59]: can we now remove the **kwargs from the arguments of DescrptSeA  ?

File: examples/water/se_e3/input_torch.json
Comment (iProzd) [reply examples/water/se_e3/input_torch.json:1]: They are the same except 'training'/'stat_file'. Yes, for clarity I agree to remove this PyTorch example.

**Comment by iProzd** [comment]
A UT is needed for testing consistency between results using `exclude_types` and results from masked nlist. I opened an issue https://github.com/deepmodeling/deepmd-kit/issues/3830 and a seperate PR needed, after finding inconsistencies in some of the existing descriptors.

File: deepmd/dpmodel/descriptor/se_t.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    DPPath,
+)
+from deepmd.utils.version import (
+    check_version_compatibility,
+)
+
+try:
+    from deepmd._version import version as __version__
+except ImportError:
+    __version__ = "unknown"
```
Comment (njzjz) [inline deepmd/dpmodel/descriptor/se_t.py:22]: Unused import.

**Review by wanghan-iapcm — APPROVED** [review]

File: deepmd/dpmodel/descriptor/se_t.py (hunk 2 above)
Comment (iProzd) [reply deepmd/dpmodel/descriptor/se_t.py:22]: I noticed that unused import of `__version__` exists **several files** in deepmd/dpmodel. Maybe I use another PR to fix them after this PR merged.

**Review by njzjz — APPROVED** [review]

**Pull request merged**